ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -97,6 +95,75 @@ public void onUnityAdsFailedToLoad(String placementId) {
     }
   };
 
+  /**
+   * IUnityAdsExtendedListener instance for call backs from show method only.
+   */
+  private IUnityAdsExtendedListener mUnityShowListener = new IUnityAdsExtendedListener() {
+    @Override
+    public void onUnityAdsClick(String s) {
+      Log.v(TAG, "Unity interstitial ad for placement ID '" + mPlacementId + "' was clicked.");
+      // Unity Ads ad clicked.
+      if (mMediationInterstitialListener == null) {
+        return;
+      }
+      mMediationInterstitialListener.onAdClicked(UnityAdapter.this);
+      // Unity Ads doesn't provide a "leaving application" event, so assuming that the
+      // user is leaving the application when a click is received, forwarding an on ad
+      // left application event.
+      mMediationInterstitialListener.onAdLeftApplication(UnityAdapter.this);
+    }
+
+    @Override
+    public void onUnityAdsPlacementStateChanged(String s, UnityAds.PlacementState placementState, UnityAds.PlacementState placementState1) {
+      // This callback is not forwarded to Google Mobile Ads SDK. onUnityAdsError should be used
+      // to forward Unity Ads SDK state to Google Mobile Ads SDK.
+    }
+
+    @Override
+    public void onUnityAdsReady(String s) {
+      // Logic to mark a placement ready has moved to the IUnityAdsLoadListener function
+      // onUnityAdsAdLoaded.
+    }
+
+    @Override
+    public void onUnityAdsStart(String s) {
+      // Unity Ads video ad started playing. Google Mobile Ads SDK does not support
+      // callbacks for Interstitial ads when they start playing.
+    }
+
+    @Override
+    public void onUnityAdsFinish(String s, UnityAds.FinishState finishState) {
+
+      // Unity Ads ad closed.
+      UnityAds.removeListener(mUnityShowListener);
+
+      if (mMediationInterstitialListener == null) {
+        return;
+      }
+
+      if (finishState == UnityAds.FinishState.ERROR) {
+        Log.v(TAG, "Unity interstitial ad for placement ID '" + mPlacementId +
+                "' finished with an error.");
+      } else {
+        Log.v(TAG, "Unity interstitial ad for placement ID '" + mPlacementId +
+                "' finished playing.");
+      }
+
+      mMediationInterstitialListener.onAdClosed(UnityAdapter.this);
+    }
+
+    @Override
+    public void onUnityAdsError(UnityAds.UnityAdsError unityAdsError, String errorMessage) {
```

Comment:
does admob not have an error callback that we need to call on the show error?

Reply:
yes, that's why we only do logging here.